LayoutTest/TestCase/LYTLayoutFailingTestSnapshotRecorder.m

```diff
@@ -0,0 +1,233 @@
+    NSString *footer = @"</TABLE></BODY></HTML>";
+    NSFileHandle *fileHandler = [NSFileHandle fileHandleForUpdatingAtPath:filePath];
+    
+    [fileHandler seekToEndOfFile];
+    [fileHandler writeData:[footer dataUsingEncoding:NSUTF8StringEncoding]];
+    [fileHandler closeFile];
+}
+
+- (void)appendViewWithWidth:(CGFloat)width
+                     height:(CGFloat)height viewData:(NSDictionary *)data
+            toLogWithFailureDescription:(NSString *)description
+             withInvocation:(NSInvocation *)invocation {
+    if (!description) {
+        description = @"";
+    }
+    
+    NSString *filePath = [self indexHTMLFilePath];
+    NSString *imagePath = [NSString stringWithFormat:@"%@/%@", [self methodNameForInvocation:invocation], [self nameForImageWithWidth:width height:height data:data]];
+    NSString *errorHTML = [NSString stringWithFormat:@"<TR><TD>%@</TD><TD><IMG src='%@' alt='No Image'></TD><TD>%@</TD></TR>", description, imagePath, data.description];
+    errorHTML = [errorHTML stringByReplacingOccurrencesOfString:@"\n" withString:@""];
+    NSFileHandle *fileHandler = [NSFileHandle fileHandleForUpdatingAtPath:filePath];
+    
+    [fileHandler seekToEndOfFile];
+    [fileHandler writeData:[errorHTML dataUsingEncoding:NSUTF8StringEncoding]];
+    [fileHandler closeFile];
+}
+
+- (NSString *)indexHTMLFilePath {
+    NSString *documentsDirectory = [self commonRootPath];
+    return [documentsDirectory stringByAppendingPathComponent:@"index.html"];
+}
+
+- (void)deleteCurrentFailingSnapshots {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    [fileManager removeItemAtPath:[self commonRootPath] error:nil];
+}
+
+- (void)createDirectoryForInvocationIfNeeded:(NSInvocation *)invocation {
+    NSString *directoryPath = [self directoryPathForCurrentInvocation:invocation];
+    BOOL isDirectory = NO;
+    if (![[NSFileManager defaultManager] fileExistsAtPath:directoryPath isDirectory:&isDirectory]) {
+        [[NSFileManager defaultManager] createDirectoryAtPath:directoryPath withIntermediateDirectories:YES attributes:nil error:nil];
+    }
+}
+
+/**
+ Returns the path to the directory to save snapshots of the current failing test. Path includes class and method name
+ e.g. {FULL_PATH}/SamepleTableViewCellLayoutTests/testSampleTableViewCell
+ */
+- (NSString *)directoryPathForCurrentInvocation:(NSInvocation *)invocation {
+    NSString *documentsDirectory = [self commonRootPath];
+    
+    NSString *directoryPath = [documentsDirectory stringByAppendingString:[NSString stringWithFormat:@"/%@", [self methodNameForInvocation:invocation]]];
+    return directoryPath;
+}
+
+- (NSString *)methodNameForInvocation:(NSInvocation *)invocation {
+    return NSStringFromSelector((SEL)[invocation selector]);
+}
+
+- (NSString *)commonRootPath {
+    return [self commonRootPathForInvocationClass:self.invocationClass];
+}
+
+- (NSString *)commonRootPathForInvocationClass:(Class)classForRoot {
```

Comment:
Oh, you already have this method. I'd take out the helper method then and just manually pass in self.invocationClass. Slightly more typing, but clearer I think